rest_registration/api/views/login.py

```diff
@@ -30,6 +30,10 @@ def login(request):
     try:
         user = login_authenticator(serializer.validated_data, serializer=serializer)
     except UserNotFound:
+        if registration_settings.USE_NON_FIELD_ERRORS_KEY_FROM_DRF_SETTINGS:
```

Comment:
I see a lot of this repeatable code in this PR. Perhaps it should be done at different level do avoid repetition and increased code complexity?

Options I see now:

1.  overriding `APIException.get_full_details` 
    https://github.com/encode/django-rest-framework/blob/79c37d0dc3740aa1c439f8a447e5e8b64fd567be/rest_framework/exceptions.py#L124-L130
    in `BadRequest` exception (should be done in Django-REST-Registration code)
2.  using custom DRF exception handing (should be done in one's application code):
    https://www.django-rest-framework.org/api-guide/exceptions/#custom-exception-handling